Fix batch actions helper & Signal payload encoding

rossedfort opened the pull request:
<!--- Note to EXTERNAL Contributors -->
<!-- Thanks for opening a PR!
If it is a significant code change, please **make sure there is an open issue** for this.
We work best with you when we have accepted the idea first before you code. -->

<!--- For ALL Contributors 👇 -->

## Description & motivation 💭 <!-- Describe what has changed in this PR and the motivation behind it -->
- This PR fixes a bug where batch actions were showing in UI's running against older temporal server versions that do not support the batch API's. The fix was to use an existing derived store, `supportsBulkActions`, which was already checking for advanced visibility and temporal server version > 1.18, but then adding the utility that also checks the client action flags from the settings API.
The requirements for seeing batch actions in the UI is now as follows:
1. temporal server version > 1.18
2. advanced visibility
3. settings.disableWriteActions = false
4. settings.batchActionsDisabled = false
5. settings.workflowCancelDisabled && settings.workflowTerminateDisabled = false
- This also addresses a small bug where Signal payloads were being encoded as a string rather than JSON.
### Screenshots (if applicable) 📸 <!-- Add screenshots or videos -->
N/a
### Design Considerations 🎨 <!-- Any questions, concerns, thoughts for Design? -->
N/a
## Testing 🧪 <!-- Describe what has changed in this PR and the motivation behind it -->

### How was this tested 👻 <!--- Please describe how you tested your changes and tests that were added -->

- [x] Manual testing
- [ ] E2E tests added
- [ ] Unit tests added

### Steps for others to test: 🚶🏽‍♂️🚶🏽‍♀️ <!--- Please describe how we can test the changes in the PR -->
- Using the docker-compose repo, set the TEMPORAL_VERSION to something older than 1.19
- run the docker containers using docker-compose up
- run the UI against the docker containers using `pnpm dev:docker`
- ensure that bulk actions are not available (even though advanced visibility and settings API flags are enabled)
## Checklists

### Draft Checklist <!-- Add todos if not ready to review -->

### Merge Checklist <!-- Add todos if not ready to merge -->

### Issue(s) closed <!-- add issue number here -->
closes #1225
## Docs

### Any docs updates needed? <!--- Update README if applicable or point out where to update docs.temporal.io -->

Alex-Tideman commented:
Just trying to grok exactly where the issue was. Did we just miss adding the settings flags to those checks for enableBulkActions?

rossedfort commented:
> Just trying to grok exactly where the issue was. Did we just miss adding the settings flags to those checks for enableBulkActions?

@Alex-Tideman I'm pretty sure it was introduced here: https://github.com/temporalio/ui/pull/1037/files#diff-ca6b214d7adfa03eee3022ad4c427567ffa4aa67c4a02f1a3d64274a8543b143 this is when we started using the settings flags, and I switched from using `$supportsBulkActions` (which checks for temporal server version > 1.18) to `bulkActionsEnabled` (which doesn't have a temporal server version check at all). 

So I just moved `bulkActionsEnabled` into the `supportsBulkActions` derived store, this way we check temporal server version, advanced visibility enabled, and settings flags.
